### PR TITLE
dev: setup script

### LINKED ISSUE #1500: dev: supercharge `make setup`
cc @enitrat for scoping

=> supercharge `make setup` with the following commands:
- `which scarb` to check asdf scarb installation is in path 
- do `foundryup` 
- asdf install

### PR DESCRIPTION
**Pull request by enitrat**
Closes #1500

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/1502)

### CODE REVIEW

**Comment by enitrat** [comment]
@Eikix can you try `make setup` with this and check all runs fine now?

File: kakarot_scripts/setup/setup.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        "katana",
+        version=KATANA_VERSION,
+    )
+
+
+def setup_local() -> None:
+    if sys.version_info < PYTHON_MIN_VERSION:
+        raise SetupError(
+            f"Python {PYTHON_MIN_VERSION[0]}.{PYTHON_MIN_VERSION[1]} or higher is required."
+        )
```
Comment (ClementWalter) [inline kakarot_scripts/setup/setup.py:89]: install version with `uv`. This script should be able to run with even the lowest versions

File: kakarot_scripts/setup/setup.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                f"echo '. $HOME/.asdf/asdf.sh' >> {shell_config}",
+                "Failed to source asdf environment",
+            )
+            print("Please restart your terminal to use asdf.")
+        else:
+            print("Please add asdf to your shell configuration manually.")
+    else:
+        print("asdf is already installed.")
+
+    if is_command_available("asdf"):
```
Comment (ClementWalter) [inline kakarot_scripts/setup/setup.py:131]:
```suggestion
```

File: kakarot_scripts/setup/setup.py
Comment (ClementWalter) [inline kakarot_scripts/setup/setup.py:1]: you don't unstall uv here?

**Review by ClementWalter — COMMENTED** [review]
I'm not against a simple script for doing this though more ad hoc tools exist. Just bear in mind then that
- the python version requirement for the script should be as low as possible so that almost anyone can run it
- only python should be required to run it and all the tools added therein (missing uv currently if I'm not mistaken)

File: kakarot_scripts/setup/setup.py
Comment (enitrat) [reply kakarot_scripts/setup/setup.py:1]: uv is needed to run this script. so no

**Comment by enitrat** [comment]
i consider that uv needs to be manually installed to run this script. perhaps its a mistake? but because it's a python script, it needs python to run anyway 🤔

**Comment by enitrat** [comment]
modified the makefile to use raw `python` and installed uv in the script.

File: kakarot_scripts/setup/setup.py
Comment (ClementWalter) [inline kakarot_scripts/setup/setup.py:1]: use a logger instead of print

**Review by ClementWalter — APPROVED** [review]
logging could be a bit nicer, and submodule updated but tested and it's already nice

**Pull request merged**